### ISSUE TITLE
Removing (programmatically) a Tab from DOM breaks keyboard (left/right) navigation of Tabs. 

### ISSUE THREAD

**Issue by mtwagner** [indiana-university/rivet-source#647]
## Bug
When programmatically removing (and presumably adding) a tab (button and panel) from a rvt_tab, left/right keyboard navigation breaks. This manifests as not being able to keyboard navigate past the position from where the tab button was removed. 

## Expected behavior
Ideally keyboard navigation would continue to work when programmatically adding or removing a tab button and panel from the DOM. 

## Relevant context (required)
Internally this is caused by the way _initElements() initiates this.tabs. It creates a new shallow copy (using Array.from) of the tab list at init time, which is then used for navigation references, detached from what is in the actual DOM during usage. In turn if the DOM is programmatically changed by user actions to add or remove tabs, those changes aren't reflected in navigation. However mouse clicks function as expected due to not relying on this internal copy of the tabs (see onClick and _getClickedTabId). 

## Steps to reproduce
document.getElementById(tab-button-id).remove()
document.getElementById(tab-panel-id).remove()
Then attempt to navigate via keyboard.

**Comment by mtwagner** [comment]
While lacking elegance, the following inserted into locally hosted rivet code does the job of updating the internal representation of tabs and panels. 
```javascript	
	init: function init() {
	.
	.
            Component.bindMethodToDOMElement(this, 'refreshTabs', this.refreshTabs);
	},
	...
	refreshTabs: function(){
            this.tabs = Array.from(this.element.querySelectorAll(this.tabSelector));	
            this.panels = Array.from(this.element.querySelectorAll(this.panelSelector));		
	},	
```

Then executing the following in a rivet app after programmatically removing a tab updates the internal tabs/panels, functionally restoring arrow key navigation: 
```javascript
	document.getElementById(tab-button-id).remove()
	document.getElementById(tab-panel-id).remove()
	document.querySelector('[data-rvt-tabs="tabset-scenarios"]').refreshTabs()
```

**Comment by scottanthonymurray** [comment]
This has been fixed in `2.2.0`.

Feel free to reopen this issue if the problem persists. Thanks!
